### PR TITLE
Remove the module_id warnings from unit tests

### PR DESCRIPTION
**Pull request by afrittoli**
Some of the test models - those defined in zip files, where not updated to the new config format with module_id, which caused a lot of warnings being logged when running unit tests.

Update the zipped models to use the module_id instead. Add a new test that specifically covers the no-module-id deprecation warning, verifying that the deprecation is raised via pytest.deprecated_call()

### CODE REVIEW

**Comment by afrittoli** [comment]
@joerunde this is the second part after https://github.com/caikit/caikit/pull/164 🙏

File: tests/core/test_model_manager.py [hunk 1]
```diff
@@ -185,6 +194,7 @@ def test_import_module_registry(self):
 TEST_DATA_PATH = os.path.join("tests", "fixtures")
 DUMMY_LOCAL_MODEL_NAME = "dummy_module_foo"
 DUMMY_BACKEND_MODEL_NAME = "dummy_module_backend"
+DUMMY_NO_ID_MODEL_NAME = "dummy_module_no_id"
```
Comment (evaline-ju) [inline tests/core/test_model_manager.py:197]: nit: this doesn't seem used - was this intended for the added test?

Comment (afrittoli) [reply tests/core/test_model_manager.py:197]: Nice catch, thank you, I defined it and forgot to use it in the test 🤦 
Fixed now

**Review by evaline-ju — APPROVED** [review]
LGTM - thanks for cleaning up the warnings!

**Pull request merged**